Ensures params.per is always set if body.per is set to prevent duplicate/missing data

Beslinda opened the pull request:
The requests with undefined `params.per` but defined `body.per` return duplicate and missing data. 

The initial request returns a default of 100 items, but in the subsequent requests the `params.per`  is set to the value of `body.per`(usually set at  25)

Beslinda committed:
Ensurea params.per is always set to prevent duplicate data when params.per but body.per is set

Beslinda committed:
Ensures params.per is always set

Beslinda committed:
Ensure both params.per and body.per are always set

mfx-jgoetzinger reviewed (commented):
thanks for contributing! however, I don't think this is the right approach. there is lots of endpoints that do not offer pagination at all. also for those that do, the backend should already apply defaults.

what's wanting you to add this?

Beslinda commented:
If we make a request with no `params.per` but with the` body.per` set to 25, the initial request will  return 100 items but the subsequent requests [here](https://github.com/mediafellows/chipmunk/pull/48/files#diff-2fcebfcceb069d1c4cf9ff41b5a758c3a64296659e84950cce8dfeca2e930dfcL27) will return results with 25 items. 

In the end when we merge all these results [here](https://github.com/mediafellows/chipmunk/pull/48/files#diff-2fcebfcceb069d1c4cf9ff41b5a758c3a64296659e84950cce8dfeca2e930dfcL37)  we get messed up data. 

I already adjusted this on the FE side and I updated the code here to not change the current logic at all but prevent this situation from happening. If `body.per` exists the `params.per` must be set as well.

```
  // Only modify opts for the action call if per was explicitly set
  const explicitPer = get(opts, "params.per") || get(opts, "body.per");
  const finalOpts = explicitPer !== undefined 
    ? { ...opts, params: { ...opts.params, per: explicitPer } }
    : opts;

  const result = await action<T>(appModel, actionName, finalOpts, config);
```

cc: @mfx-jgoetzinger

Beslinda committed:
Set per if explicitely set in body or params

mfx-jgoetzinger commented on src/unfurl.ts:
```diff
@@ -13,8 +13,18 @@ export default async <T> (
 ): Promise<IResult<T>> => {
   const per = get(opts, "params.per") || get(opts, "body.per") || 100;
+  
+  // modify opts for the action call if per was explicitly set in body or params
+  const explicitPer = get(opts, "params.per") || get(opts, "body.per");
+  const finalOpts = explicitPer !== undefined 
+    ? { 
+        ...opts, 
+        params: { ...opts.params, per: explicitPer },
+        body: { ...opts.body, per: explicitPer }
+      }
+    : opts;
```
why only `per`?

what about `page` or `sort` etc ?

mfx-jgoetzinger reviewed (approved):
sorry, this review took a long long time. don't know if it's still needed anyway
